src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2731,6 +2742,18 @@ private bool HasVariantConversionNoCycleCheck(NamedTypeSymbol source, NamedTypeS
                 destinationTypeArguments.Free();
             }
 
+            static bool isTypeIEquatable(NamedTypeSymbol type)
+            {
+                return type is
+                {
+                    IsInterface: true,
```

Comment:
>IsInterface [](start = 20, length = 11)

Also need `ContainingSymbol { Kind: SymbolKind.Namespace }` to check against nested types.